utils/cancelable.go

```diff
@@ -0,0 +1,114 @@
+package utils
+
+import "errors"
+
+// BlockingOperation type defines a function which takes no arguments and is expected to return val and error as result
+// and is expected to be blocking (i.e. some request to external system)
+type BlockingOperation func() (interface{}, error)
+
+// CleanupCallback type defines a function which is called when blocking operation returns but it was Cancelled,
+// callback parameters are val and error returned by blocking operation
+type CleanupCallback func(val interface{}, err error)
+
+// CancelChannel defines a channel which is used to monitor when blocking action is Cancelled
+type CancelChannel chan int
+
+// ErrRequestCancelled error is returned by CancelableAction.Call() method when Cancel() method was called
+var ErrRequestCancelled = errors.New("request was cancelled")
+
+// ErrUndefinedRequest error is returned by CancelableAction.Call() by default when BlockingOperation was not defined
+// ( CancelableAction.Request(BlockingOperation) was not called )
+var ErrUndefinedRequest = errors.New("undefined request called")
+
+// SkipOnError decorator takes callback function with single val as parameter and returns CleanupCallback function, which calls
+// given callback only if no error was returned by blocking operation (i.e. good when cleanup is need only when blocking operation succeded but was Cancelled before)
+func SkipOnError(callback func(interface{})) CleanupCallback {
```

Comment:
First though when reading hame is that this is empty cleanup callback - it simple does nothing (skips callback).
Maybe `InvokeOnSuccess`?

Reply:
Nice naming. Fixed